Project Atlanta UpdateDialog API and DialogUpdated Event

calvinkwtang opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

calvinkwtang committed:
Adds DialogUpdate WIP

calvinkwtang committed:
Adds DialogUpdate WIP

calvinkwtang committed:
UpdateDialog API added and finished

calvinkwtang committed:
UpdateDialog API added and finished

calvinkwtang committed:
Re-adds DialogUpdated event and moves operationcallbackuri from being a query parameter for UpdateDialog

calvinkwtang committed:
Re-adds DialogUpdated event and moves operationcallbackuri from being a query parameter for UpdateDialog

calvinkwtang committed:
Remove unused ctor

calvinkwtang committed:
Remove unused ctor

calvinkwtang committed:
Replace DialogInputType ctors with direct calls

calvinkwtang committed:
Replace DialogInputType ctors with direct calls

calvinkwtang committed:
Remove ability to set DialogInputType after ctor

calvinkwtang committed:
Remove ability to set DialogInputType after ctor

calvinkwtang committed:
Update expected response code of UpdateDialog and UpdateDialogAsync

calvinkwtang committed:
Update expected response code of UpdateDialog and UpdateDialogAsync

calvinkwtang committed:
Update swagger to latest

calvinkwtang committed:
Update swagger to latest

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Communication.CallAutomation](https://apiview.dev/Assemblies/Review/85dcaed7e707442da4ff62e68c2ab36a?revisionId=dcfe6457bab348019210865b59ec2886)

fangchen0601 committed:
add SourceCallerIdNumber to AnswerCallOptions (#41116)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240109.1 (#41105)

wiboris committed:
Updated Batch Track 2 Management plane SDK for 2023-11-01 API version (#40804)

Co-authored-by: Arcturus Zhang <dapzhang@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Batch (#41131)

christothes committed:
Preview implementation for proof of possession token support (#41015)

JoshLove-msft committed:
Prepare for release (#41144)

* Prepare for release

* Add tracing fixes to changelog

ArcturusZhang committed:
introduce mrw feature for resourcemanager, and add a shared test case to validate the mrw pattern (#41133)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240110.1 (#41139)

ArthurMa1978 committed:
Add getEntities operation for Azure.ResourceManager (#41106)

mjafferi-msft committed:
[.NET] Granular Token feature GA Post-release | Dependent SDK Recording updates (#40944)

* Updating dependent SDK's recording with latest Identity API version

azure-sdk committed:
Increment version for resourcemanager releases (#41189)

jsquire committed:
[Event Hubs] Update system property types (#41156)

* [Event Hubs] Update system property types

 The focus of these changes is to update the type of several existing
 values in the `EventData.SystemProperties` collection so that they are 
properly represented as .NET string types.  Previously, the underlying 
AMQP types were unintentionally returned, forcing callers to call
 `ToString()` to read the value.

  This is a behavioral breaking change that will impacts only those callers
  who were explicitly casting system property values to `AmqpAddress` or
  `AmqpMessageId` before calling `ToString`.

  The affected system properties are:
  - MessageId
  - CorelationId
  - To
  - ReplyTo

joseharriaga committed:
[Azure OpenAI] Remove the setter of the Functions property of ChatCompletionsOptions (#41152)

m-redding committed:
[Event Hubs] Add timeout for starting background processes in the buffered Producer (#41110)

* initial changes

* add test

* add another test

* add additional docs

* more docs

* Apply suggestions from code review

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* feedback

* add changelog entry

* fix test

* remove test that is flaky in the pipeline

* switch string to resources

---------

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7445 (#41155)

* Add package filter parameter to select recursively found packages in release

* Handle empty value for packageFilter

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.Core (#41194)

JoshLove-msft committed:
Prepare for release (#41195)

amnguye committed:
[Storage] [WebJobs] Do not pass cancelled token to ExecuteAsync when draining  (#40792)

* WIP

* WIP

* Adding DrainModeManager use in Queue Listener

* Update changelog

* Updating existing tests

* Update existing tests Part 2

calvinkwtang committed:
Removes DialogStateResponse from UpdateDialog and adds unit tests for UpdateDialog

calvinkwtang committed:
Removes DialogStateResponse from UpdateDialog and adds unit tests for UpdateDialog

RodgeFu committed:
Make Reason required for attribute CallerShouldAuditAttribute (#40682)

* make Reason required for attribute CallerShouldAudit

* update CallerShouldAuditAttribute in KeyVault

* not use name parameter

* add api compat baseline for attribute CallerShouldAuditAttribute

* add compat baseline for storage blob, file share and queue

* add compat baseline for azure.storage.files.datalake

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240111.1 (#41210)

azure-sdk committed:
Increment package version after release of Azure.Core.Experimental (#41196)

annelo-msft committed:
Add Pipeline Types to System.ClientModel library (#41085)

* initial addition of files

* pull buffering policy from this PR

* Initial tests for BinaryContent and export API

* increase code coverage

* misc tidy

* Updates from API renames

* Add message property bag tests

* backup of WIP

* Pipeline options tests and fix

* pipeline create tests

* PipelinePolicy update and test

* Add tests for StreamExtensions

* nit

* pr fb

* PR feedback on ArrayBackedPropertyBag

* pr fb

* pr fb

* pr fb

* pr fb

calvinkwtang committed:
Merge branch 'calvintang/atlantadialogupdate' of https://github.com/Azure/azure-sdk-for-net into calvintang/atlantadialogupdate
